### ISSUE TITLE
u-boot-spl.bin too big on GCC12

### ISSUE THREAD

**Issue by fmoessbauer** [electrorys/jh7110_uboot_spl#1]
Hi, thanks for creating this tool.

When running it on a `u-boot-spl.bin` that was generated using default configuration (from the SDK) but on GCC12, the created file is slightly too big (no matter if `-Os` is added).

The proprietary tool also accepts the generated blob, but this tool bails out with the size limit warning. Also the size limit is somehow misleading, as at least the input file is smaller.

```
# added this debug output
size: 129760, max 129024
File too large! Please rebuild your SPL with -Os. Maximum allowed size is 130048 bytes.
```

I did not yet check the details - maybe it is due to a different aligning or a different linker script. Anyways, I'm wondering if the created artifact of the proprietary tool is actually valid. And if so, where this limit comes from.

**Comment by strlcat** [comment]
Nope, indeed this is a bug that must be fixed, however as for latter I added size check just because of default U-Boot SPL mtd0 partition length is 128K (131072) just to avoid of overlapping. The reason behind it is that when I did full dump of my VF2 SPI flash, I found something looking very familiar to JH7110 recovery firmware at offset 0x20000. So I thought one shall never try to overwrite it even if they'll update their SPL via U-Boot.
